fendermint/vm/interpreter/src/chain.rs

```diff
@@ -89,32 +97,72 @@ where
             CheckpointPoolItem::BottomUp(ckpt) => ChainMessage::Ipc(IpcMessage::BottomUpExec(ckpt)),
         });
 
+        // Prepare top down proposals
+        match atomically_or_err(|| finality_provider.next_proposal()).await {
+            Ok(Some(proposal)) => msgs.push(ChainMessage::Ipc(IpcMessage::TopDownProposal(
+                ParentFinalityProposal {
+                    height: proposal.height as ChainEpoch,
+                    block_hash: proposal.block_hash,
+                },
+            ))),
+            Ok(None) => {
+                tracing::debug!("parent finality proposal not ready yet");
```

Comment:
Or maybe this node is root and it will never be ready, in which case this log would be just noise.

From what I understand the default polling of 10s means that with the default 2s block finality we will see this quite often. Maybe it could live inside the provider, instead of here, as it's quite normal.

Reply:
I removed this `debug`.